### PR TITLE
[fix](Nereids) fixed the limit offset error

### LINKED ISSUE #36699: [Bug] 使用开窗函数count(1) over()，当where后有limit...offset...时查询结果为空
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0.7、 2.0.11

### What's Wrong?

select语句中包含count() over()等开窗函数时，最后使用了分页查询limit xxx offset xxx，如果limit前没有order by子句，则无法查询出数据

### What You Expected?

即使不加order by , 也正常返回数据

### How to Reproduce?

create database demo;
CREATE TABLE demo.dma_emp (
  `empno` INT NULL,
  `ename` VARCHAR(255) NULL,
  `job` VARCHAR(255) NULL,
  `mgr` INT NULL,
  `hiredate` VARCHAR(255) NULL,
  `sal` FLOAT NULL,
  `comm` FLOAT NULL,
  `deptno` INT NULL
) ENGINE=OLAP
DUPLICATE KEY(`empno`, `ename`)
DISTRIBUTED BY HASH(`empno`) BUCKETS 10
;

INSERT INTO demo.dma_emp VALUES (7788, 'SCOTT', 'ANALYST', 7566, '1987-04-19', 3000, 0, 20);
INSERT INTO demo.dma_emp VALUES (7844, 'TURNER', 'SALESMAN', 7698, '1981-09-08', 1500, 0, 30);
INSERT INTO demo.dma_emp VALUES (7934, 'MILLER', 'CLERK', 7782, '1982-01-23', 1300, 0, 10);
INSERT INTO demo.dma_emp VALUES (7499, 'ALLEN', 'SALESMAN', 7698, '1981-02-20', 1600, 300, 30);
INSERT INTO demo.dma_emp VALUES (7566, 'JONES', 'MANAGER', 7839, '1981-04-02', 2975, 0, 20);
INSERT INTO demo.dma_emp VALUES (7654, 'MARTIN', 'SALESMAN', 7698, '1981-09-28', 1250, 1400, 30);
INSERT INTO demo.dma_emp VALUES (7782, 'CLARK', 'MANAGER', 7839, '1981-06-09', 2450, 0, 10);
INSERT INTO demo.dma_emp VALUES (7876, 'ADAMS', 'CLERK', 7788, '1987-05-23', 1100, 0, 20);
INSERT INTO demo.dma_emp VALUES (7698, 'BLAKE', 'MANAGER', 7839, '1981-05-01', 2850, 0, 30);
INSERT INTO demo.dma_emp VALUES (7839, 'KING', 'PRESIDENT', NULL, '1981-11-17', 5000, 0, 10);
INSERT INTO demo.dma_emp VALUES (7369, 'SMITH', 'CLERK', 7902, '1980-12-17', 800, 0, 20);
INSERT INTO demo.dma_emp VALUES (7902, 'FORD', 'ANALYST', 7566, '1981-12-03', 3000, 0, 20);
INSERT INTO demo.dma_emp VALUES (7521, 'WARD', 'SALESMAN', 7698, '1981-02-22', 1250, 500, 30);
INSERT INTO demo.dma_emp VALUES (7900, 'JAMES', 'CLERK', 7698, '1981-12-03', 950, 0, 30);

-- 1、不加order by 无法返回数据
select 
  count() over() sal_order,
  t.* 
from demo.dma_emp t
-- order by sal asc
limit 2 offset 2;

-- 2、不加order by 无法返回数据
select 
  row_number() over(order by sal desc) sal_order,
  t.* 
from demo.dma_emp t
-- order by sal asc
limit 2 offset 2;

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by felixwluo**
## Proposed changes

Issue Number: close #36699

### CODE REVIEW

File: regression-test/suites/nereids_syntax_p0/test_limit.groovy [hunk 1]
```diff
@@ -19,7 +19,11 @@ suite("test_limit") {
     sql 'set enable_nereids_planner=true'
     sql 'set enable_fallback_to_original_planner=false'
 
-    
+    def DBname = "limit_test_db" 
+    sql "DROP DATABASE IF EXISTS ${DBname}"
+    sql "CREATE DATABASE IF NOT EXISTS ${DBname}"
+    sql "use ${DBname}"
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/test_limit.groovy:25]: why create a db?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 2]
```diff
@@ -1827,9 +1827,15 @@ public PlanFragment visitPhysicalNestedLoopJoin(
 
     @Override
     public PlanFragment visitPhysicalLimit(PhysicalLimit<? extends Plan> physicalLimit, PlanTranslatorContext context) {
+        Plan childPlan = physicalLimit.child(0);
         PlanFragment inputFragment = physicalLimit.child(0).accept(this, context);
         PlanNode child = inputFragment.getPlanRoot();
-        child.setLimit(MergeLimits.mergeLimit(physicalLimit.getLimit(), physicalLimit.getOffset(), child.getLimit()));
+        if (childPlan instanceof PhysicalLimit) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1833]: the better way fix this problem
1. remove post process rule: org.apache.doris.nereids.processor.post.AddOffsetIntoDistribute
2. translate global limit into exchange, if its child is not a exchange node. refer to org.apache.doris.nereids.glue.translator.PhysicalPlanTranslator#visitPhysicalTopN

File: regression-test/suites/nereids_syntax_p0/test_limit.groovy (hunk 1 above)
Comment (felixwluo) [reply regression-test/suites/nereids_syntax_p0/test_limit.groovy]: done,It has been deleted

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java (hunk 2 above)
Comment (felixwluo) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java]: thx, done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 3]
```diff
@@ -1829,9 +1829,17 @@ public PlanFragment visitPhysicalNestedLoopJoin(
     public PlanFragment visitPhysicalLimit(PhysicalLimit<? extends Plan> physicalLimit, PlanTranslatorContext context) {
         PlanFragment inputFragment = physicalLimit.child(0).accept(this, context);
         PlanNode child = inputFragment.getPlanRoot();
-        child.setLimit(MergeLimits.mergeLimit(physicalLimit.getLimit(), physicalLimit.getOffset(), child.getLimit()));
-        // TODO: plan node don't support limit
-        // child.setOffset(MergeLimits.mergeOffset(physicalLimit.getOffset(), child.getOffset()));
+
+        if (!(inputFragment.getPlanRoot() instanceof ExchangeNode)) {
+            inputFragment.getPlanRoot().setLimit(physicalLimit.getLimit());
+            inputFragment.getPlanRoot().setOffset(physicalLimit.getOffset());
```
Comment (lide-reed) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1835]: Here the  inputFragment.getPlanRoot() should replace with the varaible child which defined above.

Comment (felixwluo) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java]: thx, done

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run compile

**Comment by felixwluo** [comment]
run p0

**Comment by felixwluo** [comment]
run cloud_p0

**Comment by felixwluo** [comment]
run cloud_p1

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 4]
```diff
@@ -1829,9 +1829,17 @@ public PlanFragment visitPhysicalNestedLoopJoin(
     public PlanFragment visitPhysicalLimit(PhysicalLimit<? extends Plan> physicalLimit, PlanTranslatorContext context) {
         PlanFragment inputFragment = physicalLimit.child(0).accept(this, context);
         PlanNode child = inputFragment.getPlanRoot();
-        child.setLimit(MergeLimits.mergeLimit(physicalLimit.getLimit(), physicalLimit.getOffset(), child.getLimit()));
-        // TODO: plan node don't support limit
-        // child.setOffset(MergeLimits.mergeOffset(physicalLimit.getOffset(), child.getOffset()));
+
+        if (!(child instanceof ExchangeNode)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1833]: 1. remove post process: AddOffsetIntoDistribute
2. in translator
2.1. if limit is local, just merge it with child.getLimit() and set it to child.limit
2.2. if limit is global, check wether it's child is exchange. if child is exchange, just merge limit and offset with exchange's limit and offset. if child is not exchange, create exchange node and set limit and offset to it

**Comment by felixwluo** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java (hunk 4 above)
Comment (felixwluo) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java]: done

**Review by morrySnow — DISMISSED** [review]

**Comment by morrySnow** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run p0

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

**Comment by felixwluo** [comment]
run buildall

**Comment by morrySnow** [comment]
run feut

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**